{{cookiecutter.project_name}}/docs/template_usage.md

```diff
@@ -142,78 +142,99 @@ Once authorized, pre-commit.ci should automatically be activated.
+
+-   [ruff][]: Many configurable checks for Python code, see [Overview of Ruff checks](#overview-of-ruff-checks).
+-   [pre-commit-hooks](https://github.com/pre-commit/pre-commit-hooks): generic pre-commit hooks for text files.
     -   **detect-private-key**: checks for the existence of private keys.
     -   **check-ast**: check whether files parse as valid python.
-    -   **end-of-file-fixer**:check files end in a newline and only a newline.
+    -   **end-of-file-fixer**: check files end in a newline and only a newline.
     -   **mixed-line-ending**: checks mixed line ending.
     -   **trailing-whitespace**: trims trailing whitespace.
     -   **check-case-conflict**: check files that would conflict with case-insensitive file systems.
--   [pyupgrade](https://github.com/asottile/pyupgrade):
+-   **forbid-to-commit**: Make sure that `*.rej` files cannot be commited.
+    These files are created by the [automated template sync](#automated-template-sync)
+    if there's a merge conflict and need to be addressed manually.
+
+[ruff]: https://beta.ruff.rs/docs/
+
+#### Overview of Ruff checks
+
+[Ruff][] implements several checks for Python code and docstrings.
+Therefore each check below mentions a the rule prefix that enables it when added to `pyproject.toml`.
+E.g. `isort` is enabled with:
+
+```toml
+[tool.ruff]
+select = [..., "I", ...]
+```
+
+The following checks are for code style and format:
+
+-   [isort](https://beta.ruff.rs/docs/rules/#isort-i) (`I`):
+    sort module imports into sections and types.
+-   [pydocstyle](https://beta.ruff.rs/docs/rules/#pydocstyle-d) (`D`):
+    pydocstyle extension of flake8.
+-   [flake8-tidy-imports](https://beta.ruff.rs/docs/rules/#flake8-tidy-imports-tid) (`TID`):
+    tidy module imports.
+-   [flake8-comprehensions](https://beta.ruff.rs/docs/rules/#flake8-comprehensions-c4) (`C4`):
+    write better list/set/dict comprehensions.
+-   [pyupgrade](https://beta.ruff.rs/docs/rules/#pyupgrade-up) (`UP`):
     upgrade syntax for newer versions of the language.
--   **forbid-to-commit**: Make sure that `*.rej` files cannot be commited. These files are created by the
-    [automated template sync](#automated-template-sync) if there's a merge conflict and need to be addressed manually.
+
+The following checks are for errors and inconsistencies:
+
+-   [pyflakes](https://beta.ruff.rs/docs/rules/#pyflakes-f) (`F`):
+    various checks for errors.
+-   [pycodestyle](https://beta.ruff.rs/docs/rules/#pycodestyle-e-w) (`E`, `W`):
+    various checks for errors.
+-   [flake8-bugbear](https://beta.ruff.rs/docs/rules/#flake8-bugbear-b) (`B`):
+    find possible bugs and design issues in program.
+-   [flake8-blind-except](https://beta.ruff.rs/docs/rules/#flake8-blind-except-ble) (`BLE`):
+    checks for blind, catch-all `except` statements.
+-   [Ruff-specific rules](https://beta.ruff.rs/docs/rules/#ruff-specific-rules-ruf) (`RUF`):
+    -   `RUF100`: remove unneccesary `# noqa` comments.
```

Comment:
We were using yesqa before. This is Ruff’s rule implementing the same functionality as yesqa. So I don’t think this PR changes functionality

Ruff will leave alone codes listed in [`external = [...]`](https://beta.ruff.rs/docs/settings/#external)

Reply:
Alright. I guess I'm a bit on the fence then.

From my understanding yesqa could be configured somewhat separately from the pre-commit linting run, it could recognize a greater set of issues than you'd necessarily want to prevent commits. Would this remove any `noqa` that doesn't prevent a commit?

Reply:
> Ruff will leave alone codes listed in [external = [...]](https://beta.ruff.rs/docs/settings/#external).

Yes, please do that. I think that should be fine.